### PR TITLE
Add `ignoredTransactions` option to filter out transactions by name

### LINKED ISSUE #3840: Filtering healthcheck transactions with sentry-opentelemetry-agent V8
### Problem Statement

I want to filter out my healthcheck transactions (`grpc.health.v1.Health/Check`, `PING`), preferably configured using an Environment variable.

I am using the new version of the sentry-opentelemetry-agent which works great, my only issue is filtering out healthcheck transactions.

### Solution Brainstorm

In a perfect world I could provide an environment variable with a list of transactions to exclude (`SENTRY_EXCLUDE_TRANSACTIONS=grpc.health.v1.Health/Check,PING`) but I would settle for being able to configure it in code while still being able to take advantage of the auto initialisation (I've tried calling Sentry.init and just specifying a `beforeSendTransaction` options but I get the error "DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.").

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Add `ignoredTransactions` option to filter out transactions by name

- can be used via ENV vars, e.g. `SENTRY_IGNORED_TRANSACTIONS=POST /person/,GET /pers.*`
- can also be set in options directly, e.g. `options.setIgnoredTransactions(...)`
- can also be set in `sentry.properties`, e.g. `ignored-transactions=POST /person/,GET /pers.*`
- can also be set in Spring config `application.properties`, e.g. `sentry.ignored-transactions=POST /person/,GET /pers.*`

## :bulb: Motivation and Context

Fixes #3840

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -774,6 +774,23 @@ public void captureSession(final @NotNull Session session, final @Nullable Hint
...
+              SentryLevel.DEBUG,
+              "Transaction was dropped as transaction name %s is ignored",
+              transaction.getTransaction());
+      options
+          .getClientReportRecorder()
+          .recordLostEvent(DiscardReason.EVENT_PROCESSOR, DataCategory.Transaction);
+      options
+          .getClientReportRecorder()
+          .recordLostEvent(
+              DiscardReason.EVENT_PROCESSOR, DataCategory.Span, transaction.getSpans().size() + 1);
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/SentryClient.java:790]: What does the +1 here do? Is that to account for the transaction?

File: sentry/src/main/java/io/sentry/util/TracingUtils.java [hunk 2]
```diff
@@ -116,4 +117,32 @@ public TracingHeaders(
       return baggageHeader;
     }
   }
+
+  /** Checks if a transaction has been ignored. */
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/util/TracingUtils.java:121]: (l) Not 100% sure about the comment here. Maybe something like this?
```suggestion
  /** Checks if a transaction is to be ignored. */
```

File: sentry/src/test/java/io/sentry/SentryClientTest.kt [hunk 3]
```diff
@@ -819,6 +819,28 @@ class SentryClientTest {
         )
     }
 
+    @Test
+    fun `transaction dropped by ignoredTransactions is recorded`() {
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/SentryClientTest.kt:823]: Maybe also add a regex ignored transaction test

File: sentry/src/main/java/io/sentry/SentryClient.java (hunk 1 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryClient.java:790]: Yes, same as further down.

**Review by lbloder — APPROVED** [review]

**Pull request merged**